src/app/insiders/insiders.component.ts

```diff
@@ -0,0 +1,75 @@
+import {AfterViewInit, Component, OnInit, ViewChild} from '@angular/core';
+import {StockDataService} from "../services/stock-data.service";
+import {InsiderData} from '../model/InsiderData';
+import {MatSort, Sort} from "@angular/material/sort";
+import {MatTableDataSource} from "@angular/material/table";
+
+@Component({
+  selector: 'insiders',
+  templateUrl: './insiders.component.html',
+  styleUrls: ['./insiders.component.css'],
+})
+export class InsidersComponent implements OnInit, AfterViewInit {
+
+  public stockName: string;
+  public data = new MatTableDataSource<InsiderData>([]);
+
+  @ViewChild(MatSort,{ static: true }) sort: MatSort;
+
+  displayedColumns: string[] = [
```

Comment:
Right order is:
`
'title',
'symbol',
'filingDate',
'reportingName',
'transactionDate',
'transactionCode',
'amount',
'sharesHeld',
'price',
'ownership',
'relationship'
`